frontend/server/src/SecurityTools.php

```diff
@@ -283,24 +283,36 @@ public static function getDecodedCloneCourseToken(
         $parser = \ParagonIE\Paseto\Parser::getLocal(
             self::getCourseCloneSecretKey(),
             \ParagonIE\Paseto\ProtocolCollection::v2()
-        )
-            ->addRule(
-                new \ParagonIE\Paseto\Rules\ValidAt(
-                    (new \DateTime())->setTimestamp(\OmegaUp\Time::get())
-                )
-            )
-            ->addRule(new \OmegaUp\ClaimRule('course', $courseAlias))
-            ->addRule(new \OmegaUp\ClaimRule('permissions', 'clone'));
-        try {
-            $parsedToken = $parser->parse($token, /*$skipValidation=*/true);
-        } catch (\ParagonIE\Paseto\Exception\PasetoException $e) {
-            throw new \OmegaUp\Exceptions\TokenDecodeException();
-        }
+        );
+        $parsedToken = $parser->parse($token, /*$skipValidation=*/true);
         /** @var array<string, string> */
         $claims = $parsedToken->getClaims();
-        if (!$parser->validate($parsedToken)) {
-            throw new \OmegaUp\Exceptions\TokenDecodeException(
-                'tokenDecodeFailed',
+        if (
+            !(new \OmegaUp\ClaimRule(
+                'permissions',
+                'clone'
+            ))->isValid(
+                $parsedToken
+            ) ||
+            !(new \OmegaUp\ClaimRule(
+                'course',
+                $courseAlias
+            ))->isValid(
+                $parsedToken
+            )
+        ) {
+            throw new \OmegaUp\Exceptions\TokenValidateException(
+                'tokenDecodeInvalid',
```

Comment:
```suggestion
                'token_invalid',
```

así no hay necesidad de afuera checar si esto es una cadena y convertirlo a otra: puedes poner la razón directamente aquí y sacarlo de la excepción en la función que lo llama.

Reply:
Listo